### ISSUE TITLE
Missing package: cryptography

### ISSUE THREAD

**Issue by MichaIng** [piwheels/packages#264]
### Package name

cryptography

### Package version

36.0.0

### PyPI URL

https://pypi.org/project/cryptography/

### piwheels URL

https://www.piwheels.org/project/cryptography/

### Python version

- [x] Python 3.5
- [X] Python 3.7
- [ ] Python 3.9

### I am the maintainer

- [ ] Yes

### More information

The recent build for Buster failed. Since the module build requires Rust now which significantly increases build times build dependencies, it would beneficial to have it pre-compiled also for Buster for a while. Raspberry Pi Bullseye has been officially released, but Raspberry Pi Buster is still the most used version in the fields.

**Comment by bennuttall** [comment]
It's failed on Buster but I'm not sure why. It's got the same configuration as Bullseye (rustc, cargo, same pip, same setuptools_rust).

Here's the end of the build output for the failure: https://paste.debian.net/1221020/
Compared to the output for the successful build on Bullseye: http://piwheels.org/logs/0000/0739/4023.txt.gz

If you can manage to build it successfully, let's see if we can narrow down what's wrong.

**Comment by MichaIng** [comment]
The `setuptools` version is not the same if I see it right: `setuptools-59.2.0` vs `57.4.0` on the failing build. Not sure about the `wheel` module version? I.e. `pip3 install setuptools==59.2.0 wheel==0.37.0` to rule out that those versions are the reason.

As you use `cargo`/`rustc` from the Raspbian repository (?), their version is different as well, but the failure is on Python level where compiling did not even start yet.

**Comment by MichaIng** [comment]
Ah no, it is indeed `cargo` which fails with a segmentation fault:
```
subprocess.CalledProcessError: Command '['cargo', 'metadata', '--manifest-path', 'src/rust/Cargo.toml', '--format-version', '1']' died with <Signals.SIGSEGV: 11>.
```
I was able to replicate as well with latest `setuptools` and `wheels`. Probably there is no way around rustup to get a more recent version of cargo and/or the Rust compiler.

**Comment by MichaIng** [comment]
It works with rustup. I saw your comment [here](https://github.com/piwheels/packages/issues/188#issuecomment-778829130) why you use the Debian packages, but actually rustup works just fine with any user, especially when it is meant only for a single user or even a single build only. It will be installed into the users home directory, and a `source $HOME/.cargo/env` will load the executables into `PATH`.

rustup is only complicated when one wants to have a single system-wide rustup/compiler/cargo instance, used by multiple users which then only have their own cargo environment. The issue on the Rust repo explains a way, but indeed there is no guarantee that this works reliable for all cases, as rustup is simply (and IMHO sadly) not designed to be used that way. But for a build server all this shouldn't play any role.

**Comment by bennuttall** [comment]
I've just built it successfully on a Pi at home by just updating pip, setuptools, etc - but I still haven't got it working on piwheels. Anyway, I've imported that build so it's there now.

**Comment by MichaIng** [comment]
This did succeed with cargo + rustc from Raspbian Buster repository? In my case the exact required step to get it working is to replace those with rustup 🤔.

**Comment by bennuttall** [comment]
Actually it looks like I had previously installed rustup on that Pi - didn't realise

**Comment by MichaIng** [comment]
That would explain it 🙂. Btw, no change to get a wheel for Python 3.5/Stretch, is it?

**Comment by bennuttall** [comment]
cryptography requires: Python >=3.6

**Comment by MichaIng** [comment]
Ah right, latest version which supports Python 3.5 is v3.1.1 and a wheel for it is there already.

**Comment by MichaIng** [comment]
Issue solved, but now the next build failed for v36.0.1 Python 3.7 🙈, reported here: #267
